src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/UnmergeService.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.digital.hmpps.personrecord.service
+
+import jakarta.transaction.Transactional
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.runBlocking
+import org.hibernate.exception.ConstraintViolationException
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.dao.CannotAcquireLockException
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.orm.ObjectOptimisticLockingFailureException
+import org.springframework.orm.jpa.JpaObjectRetrievalFailureException
+import org.springframework.orm.jpa.JpaSystemException
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.personrecord.model.person.Person
+import uk.gov.justice.digital.hmpps.personrecord.service.MergeService.Companion.MAX_ATTEMPTS
+import uk.gov.justice.digital.hmpps.personrecord.service.RetryExecutor.runWithRetry
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.CPR_SELF_MATCH
+
+@Service
+class UnmergeService(
+  private val telemetryService: TelemetryService,
+  private val matchService: MatchService,
+  @Value("\${retry.delay}") private val retryDelay: Long,
+) {
+
+  private val retryExceptions = listOf(
```

Comment:
opportunity for code reuse here? This list seems to crop up a lot

Reply:
Move to a reusable component